packages/babel-helper-builder-react-jsx-experimental/src/index.js

```diff
@@ -0,0 +1,739 @@
+import esutils from "esutils";
+import * as t from "@babel/types";
+import { addNamed, addNamespace, isModule } from "@babel/helper-module-imports";
+
+export function helper(babel, options) {
+  const FILE_NAME_VAR = "_jsxFileName";
+
+  const JSX_SOURCE_ANNOTATION_REGEX = /\*?\s*@jsxImportSource\s+([^\s]+)/;
+  const JSX_RUNTIME_ANNOTATION_REGEX = /\*?\s*@jsxRuntime\s+([^\s]+)/;
+
+  const JSX_ANNOTATION_REGEX = /\*?\s*@jsx\s+([^\s]+)/;
+  const JSX_FRAG_ANNOTATION_REGEX = /\*?\s*@jsxFrag\s+([^\s]+)/;
+
+  // This is the number of possible import names
+  // development: jsxDEV, Fragment, createElement
+  // production: jsx, jsxs, Fragment, createElement
+  const IMPORT_NAME_SIZE = options.development ? 3 : 4;
+
+  const {
+    importSource: IMPORT_SOURCE_DEFAULT = "react",
+    runtime: RUNTIME_DEFAULT = "automatic",
+    pragma: PRAGMA_DEFAULT = "React.createElement",
+    pragmaFrag: PRAGMA_FRAG_DEFAULT = "React.Fragment",
+  } = options;
+
+  return {
+    JSXNamespacedName(path, state) {
+      const throwIfNamespace =
+        state.opts.throwIfNamespace === undefined
+          ? true
+          : !!state.opts.throwIfNamespace;
+      if (throwIfNamespace) {
+        throw path.buildCodeFrameError(
+          `Namespace tags are not supported by default. React's JSX doesn't support namespace tags. \
+You can turn on the 'throwIfNamespace' flag to bypass this warning.`,
```

Comment:
It seems clearer to me that we just say what our users should do
```suggestion
You can set `throwIfNamespace: false` to bypass this warning.`,
```

Reply:
Sounds good. This warning was taken from the original Babel transform so I'll update that one as well!